add an automatic table of io supports

ahmadtourei opened the pull request:
<!--
Thanks for contributing to DASCore, community contributions are most welcomed!

Before contributing, please read through the [contributors doc](https://dascore.org/contributing/contributing.html)

Before making big changes to the code or adding large complex features, it is a good idea to
[open a discussion](https://github.com/DASDAE/dascore/discussions). Don't hesitate to ask a question or for
help if something isn't clear.
-->

## Description
Adding an automatically generated table of supported formats to the DASCore's homepage documentation.
Related to #159
<!--
Please describe your PR here. What problem are you trying to solve, or what feature are you adding?

Also link any relevant issues/discussions (this can be done using the issue/discussion number preceded by a
pound sign, e.g. `#12` without the backticks)
-->

## Checklist

I have (if applicable):

- [ ] referenced the GitHub issue this PR closes.
- [ ] documented the new feature with docstrings or appropriate doc page.
- [ ] included a test. See [testing guidelines](https://dascore.org/contributing/testing.html).
- [ ] your name has been added to the contributors page (docs/contributors.md).
- [ ] added the "ready_for_review" tag once the PR is ready to be reviewed.

d-chambers commented:
Nice @ahmadtourei,

Two suggestions:

1. Mind adding a test for this? It should go in [`test_io_core.py`](https://github.com/DASDAE/dascore/blob/master/tests/test_io/test_io_core.py). It could be as simple as generate the table, assert it is a dataframe and that its len is not 0. 

2. For the documentation page, what would you think about swapping out `True` and `False` for ✅  and  ❌? You could just use a [`replace`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.replace.html) after loading the dataframe in the quarto code block.

ahmadtourei commented:
> Nice @ahmadtourei,
> 
> Two suggestions:
> 
> 1. Mind adding a test for this? It should go in [`test_io_core.py`](https://github.com/DASDAE/dascore/blob/master/tests/test_io/test_io_core.py). It could be as simple as generate the table, assert it is a dataframe and that its len is not 0.
> 2. For the documentation page, what would you think about swapping out `True` and `False` for ✅  and  ❌? You could just use a [`replace`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.replace.html) after loading the dataframe in the quarto code block.

Thanks for the suggestions! I added a test and changed True and False to the emojis.
